test/unit/traversing.js

```diff
@@ -743,6 +743,49 @@ QUnit.test( "contents()", function( assert ) {
 	assert.equal( c[ 0 ].nodeValue, "hi", "Check node,textnode,comment contents is just the one from span" );
 } );
 
+QUnit.test( "contents() for <template />", function( assert ) {
+	assert.expect( 6 );
+
+    jQuery( "#qunit-fixture" ).append(
+        "<template id=\"template\">" +
+        "    <div id=\"template-div0\">" +
+        "        <span>Hello, Web Component!</span>" +
+        "    </div>" +
+        "    <div id=\"template-div1\"></div>" +
+        "    <div id=\"template-div2\"></div>" +
+        "</template>" +
+        "<form id=\"test_contents\">" +
+        "    <input type=\"text\" name=\"contentDocument\" />" +
+        "    <input type=\"text\" name=\"content\" />" +
+        "</form>"
+    );
+
+    var c = jQuery( "#template" ).contents();
+    assert.equal( c.length, 6, "Check template element contents" );
+
+    assert.equal( c.find( "span" ).text(), "Hello, Web Component!", "Find span in Template and check its text" );
+
+    jQuery( "<div id=\"templateTest\" />" ).append(
+        jQuery( jQuery.map( c, function( node ) {
+            return document.importNode( node, true );
+        } ) )
+    ).appendTo( document.body );
```

Comment:
You probably responding to https://github.com/jquery/jquery/pull/3462#pullrequestreview-14458118 comment though?